src/components/atoms/InputField/InputField.scss

```diff
@@ -0,0 +1,51 @@
+@import "scss/constants";
+
+$icon-width: 2.5rem;
+
+.input-field {
+  position: relative;
+
+  &--icon-start {
+    input {
+      padding-left: $icon-width;
+    }
+  }
+
+  &--icon-end {
+    input {
+      padding-right: $icon-width;
+    }
+  }
+
+  &__input {
+    height: 100%;
+
+    &:focus + .input-field__icon > * {
+      opacity: 0.6;
+    }
+  }
+
+  &__icon {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: absolute;
+    top: 0;
+    height: 100%;
+    width: $icon-width;
+    z-index: z(input-icon);
+
+    &--start {
+      left: 0;
+    }
+
+    &--end {
+      right: 0;
+    }
+
+    & > * {
+      opacity: 0.2;
+      width: 100%;
+    }
```

Comment:
same as above ^

Reply:
please, check my comment above. If we decide to remove `*` for opacity, I'll do that here as well.

hmm :thinking:, and about `width`. I think we need `*` for width property as we don't know anything about custom icons which we can assign to `iconStart`/`iconEnd` (nor className, nor tag), but we know that its width shouldn't be bigger than input-padding. Although, I should better use `max-width` property instead here.

Reply:
change proprty from `width` to `max-width`: https://github.com/sparkletown/sparkle/pull/1306/commits/70022e4294d880248ca128366bff475348bf4ab8

Reply:
I got the logic. Makes sense